docs/source/core_concepts/glossary.rst

```diff
@@ -250,6 +258,48 @@ Transaction status set
  - STATEFUL_VALIDATION_SUCCESS: the transaction has successfully passed stateful validation.
  - COMMITTED: the transaction is the part of a block, which gained enough votes and is in the block store at the moment.
 
+Pending Transactions
+^^^^^^^^^^^^^^^^^^^^
+
+.. TODO igor-egorov, 2018-07-04, IR-1356, add here a link to MST docs
+
+Any transaction that has lesser signatures at the moment than `quorum`_ of transaction creator account is considered as pending.
+Pending transaction will be submitted for stateful validation as soon as multi signature mechanism (*to be documented*)
+will collect required amount of signatures for quorum.
+
+Transaction that already has quorum of signatures can also be considered as pending in cases
+when the transaction is a part of `batch of transactions`_ and there is a not fully signed transaction.
+
+Batch of Transactions
+=====================
+
+*The feature is to be released.*
+
+Transactions batch is a feature that allows sending several transactions to Iroha at once preserving their order.
+
+Each transaction within a batch includes batch meta information.
+Batch meta contains batch type identifier (atomic or ordered) and a list of `reduced hashes <#reduced-transaction-hash>`_
+of all transactions within a batch.
+The order of hashes prescribes transactions sequence.
+
+.. TODO igor-egorov, 2018-07-04, IR-1356, add here a link to MST docs
+
+Batch can contain transactions created by different accounts.
+Any transaction within a batch can require single or multiple signatures (depends on quorum set for an account of transaction creator).
+At least one transaction inside a batch should have at least one signature to let the batch pass `stateful validation`_.
+
+Atomic Batch
+------------
+
+All the transactions within an atomic batch should pass `stateful validation`_ for the batch to be applied to a ledger.
```

Comment:
Please add sentence saying that order of transactions in atomic batch is preserved